src/symbolic.ml

```diff
@@ -129,28 +124,31 @@ struct
     include Symbolic_memory.M
 
     let concretise a =
+      let open Choice in
       let open Expr in
-      Choice.with_thread (fun thread ->
-          let (S (solver_mod, solver)) = Thread.solver thread in
-          let module Solver = (val solver_mod) in
-          match a.node.e with
-          | Val _ | Ptr (_, { node = { e = Val _; _ }; _ }) -> a
-          | Ptr (base, offset) ->
-            (* TODO: can we remove this check? We should be in a SAT branch *)
-            assert (Solver.check solver @@ Thread.pc thread);
-            let concrete_offest = Solver.get_value solver offset in
-            let cond = Relop (Eq, offset, concrete_offest) @: a.node.ty in
-            (* TODO: this should go to the pc *)
-            Solver.add solver [ cond ];
-            Ptr (base, concrete_offest) @: a.node.ty
-          | _ ->
-            (* TODO: can we remove this check? We should be in a SAT branch *)
-            assert (Solver.check solver @@ Thread.pc thread);
-            let concrete_addr = Solver.get_value solver a in
-            let cond = Relop (Eq, a, concrete_addr) @: a.node.ty in
-            (* TODO: this should go to the pc *)
-            Solver.add solver [ cond ];
-            concrete_addr )
+      let* thread = get_thread in
+      let* (S (solver_mod, solver)) = get_solver in
```

Comment:
`let+` (and you can remove the `return` below)

Reply:
Also, if you rename `get_thread` to `thread` you'll be able to simply write `let* thread in`